### ISSUE TITLE
Support MERGE functionality for links

### ISSUE THREAD

**Issue by ymark** [isl/x3ml#97]
In some cases, it is required to describe the path of a link and if this holds (i.e. if there is the corresponding part in the input that satisfies the path) we would like to skip the generation of the resulting range. 

The requirements, that we have so far for this case, is if this happens, we would like to reuse the instance that has been generated for the domain of the particular mapping.

We call this type of functionality MERGE functionality. More specifically we use the special keyword **MERGE** to declare such case. This keyword is being used inside the _relationship_ element and should appear in upper case. The following X3ML mapping describes how it is used.

```
<mapping>
	<domain>
		<source_node>//object</source_node>
		<target_node>
			<entity>
				<type>crm:E22_Man-Made_Object</type>
				<instance_generator name="UUID"/>
			</entity>
		</target_node>
	</domain>
	<link>
		<path>
			<source_relation>
				<relation>object_part</relation>
			</source_relation>
			<target_relation>
				<relationship>MERGE</relationship>
			</target_relation>
		</path>
		<range>
			<source_node>object_part</source_node>
			<target_node>
				<entity>
					<type>crm:E22_Man-Made_Object</type>
						<instance_generator name="UUID"/>
				</entity>
			</target_node>
		</range>
	</link> 
<mapping>
```

The above mapping declares that if there are elements with name _object_ having a child element with name _object_part_ then they will be assigned the same identifier. A set of resources examples describing how this functionality works will be added in the issue (as soon as it is resolved)

**Comment by ymark** [comment]
The attached files describe how the new feature is expected to function. 
[mappings.x3ml.txt](https://github.com/isl/x3ml/files/1287167/mappings.x3ml.txt)
[expected_output.rdf.txt](https://github.com/isl/x3ml/files/1287169/expected_output.rdf.txt)
[input.xml.txt](https://github.com/isl/x3ml/files/1287168/input.xml.txt)

**Comment by ymark** [comment]
The MERGE facility has been implemented in cbff42e296cb1c92810a1618ec837b825539e41b

**Comment by ymark** [comment]
The new functionality will be made available from 1.8.2
